rllib/agents/tests/test_trainer.py

```diff
@@ -258,40 +258,68 @@ def test_space_inference_from_remote_workers(self):
 
         env = gym.make("CartPole-v0")
 
-        config = pg.DEFAULT_CONFIG.copy()
-        config["env"] = "CartPole-v0"
-        config["num_workers"] = 1
+        config = (
+            pg.PGConfig()
+            .rollouts(num_rollout_workers=1, validate_workers_after_construction=False)
+            .environment(env="CartPole-v0")
+        )
 
         # No env on driver -> expect longer build time due to space
         # lookup from remote worker.
         t0 = time.time()
-        trainer = pg.PGTrainer(config=config)
+        trainer = config.build()
         w_lookup = time.time() - t0
         print(f"No env on learner: {w_lookup}sec")
         trainer.stop()
 
         # Env on driver -> expect shorted build time due to no space
         # lookup required from remote worker.
-        config["create_env_on_driver"] = True
+        config.create_env_on_local_worker = True
         t0 = time.time()
-        trainer = pg.PGTrainer(config=config)
+        trainer = config.build()
         wo_lookup = time.time() - t0
         print(f"Env on learner: {wo_lookup}sec")
         self.assertLess(wo_lookup, w_lookup)
         trainer.stop()
 
         # Spaces given -> expect shorter build time due to no space
         # lookup required from remote worker.
-        config["create_env_on_driver"] = False
-        config["observation_space"] = env.observation_space
-        config["action_space"] = env.action_space
+        config.create_env_on_driver = False
+        config.environment(
+            observation_space=env.observation_space,
+            action_space=env.action_space,
+        )
         t0 = time.time()
-        trainer = pg.PGTrainer(config=config)
+        trainer = config.build()
         wo_lookup = time.time() - t0
         print(f"Spaces given manually in config: {wo_lookup}sec")
         self.assertLess(wo_lookup, w_lookup)
         trainer.stop()
 
+    def test_worker_validation_time(self):
+        """Tests the time taken by `validate_workers_after_construction=True`."""
+        config = pg.PGConfig()
+        config.env = "CartPole-v0"
+        config.validate_workers_after_construction = True
```

Comment:
We should ...
We have NOT yet started to completely clean up everything to match the new TrainerConfig API.
This will be done as a next step, once all algos have their own config class.